### PR TITLE
SW-749 Upgrade code level to PHP 7.4

### PR DESCRIPTION
**Pull request by TobiasGraml11**

### CODE REVIEW

File: src/Core/Content/Category/SalesChannel/CachedCategoryRoute.php [hunk 1]
```diff
@@ -44,13 +38,13 @@ public function getDecorated(): AbstractCategoryRoute
...
         Request $request,
-        SalesChannelContext $salesChannelContext
+        SalesChannelContext $context
     ): CategoryRouteResponse {
-        $this->initializePluginState($salesChannelContext, $request);
+        $this->initializePluginState($context, $request);
 
-        return $this->getDecorated()->load($categoryId, $request, $salesChannelContext);
+        return $this->getDecorated()->load($navigationId, $request, $context);
     }
```
Comment (TobiasGraml11) [inline src/Core/Content/Category/SalesChannel/CachedCategoryRoute.php:48]: Argument names were updated in `AbstractCategoryRoute`

File: src/Findologic/Response/ResponseParser.php [hunk 2]
```diff
@@ -21,20 +21,11 @@
...
-     * @var ServiceConfigResource|null
-     */
-    protected $serviceConfigResource;
+    protected ?ServiceConfigResource $serviceConfigResource;
 
-    /**
-     * @var Config
-     */
-    protected $config;
+    protected ?Config $config;
```
Comment (TobiasGraml11) [inline src/Findologic/Response/ResponseParser.php:28]: This was wrongly typed, as the constructor argument is also nullable.

File: src/Findologic/Response/Xml21/Filter/ColorPickerFilter.php [hunk 3]
```diff
@@ -6,6 +6,5 @@
 
 class ColorPickerFilter extends Filter
 {
-    /** @var string */
-    protected $displayType = 'color';
+    protected ?string $displayType = 'color';
```
Comment (TobiasGraml11) [inline src/Findologic/Response/Xml21/Filter/ColorPickerFilter.php:9]: Type needs to be the same as the parent, therefore I set it nullable

File: src/Findologic/Config/FinSearchConfigEntity.php [hunk 4]
```diff
@@ -13,35 +13,18 @@ class FinSearchConfigEntity extends Entity
...
-     */
-    protected $configurationKey;
-
-    /**
-     * @var mixed
-     */
+    protected string $configurationKey;
+
+    /** @var mixed */
     protected $configurationValue;
```
Comment (msilinskas) [inline src/Findologic/Config/FinSearchConfigEntity.php:19]: Couldn't you declare types for this variable?

Comment (TobiasGraml11) [reply src/Findologic/Config/FinSearchConfigEntity.php:19]: It can basically be any type, that's why I kept using `mixed`

Comment (TobiasGraml11) [reply src/Findologic/Config/FinSearchConfigEntity.php:19]: `mixed` is not available as variable type and union types are still not supported (`array|int|bool|string`)

**Pull request merged**